lang/java/reef-common/src/main/java/org/apache/reef/runtime/common/driver/EvaluatorRequestorImpl.java

```diff
@@ -70,26 +70,27 @@ public synchronized void submit(final EvaluatorRequest req) {
     if (req.getNumber() <= 0) {
       throw new IllegalArgumentException("Given an unsupported number of evaluators: " + req.getNumber());
     }
+    if (req.getNodeNames() == null) {
+      throw new IllegalArgumentException("Node names cannot be null");
+    }
+    if (req.getRackNames() == null) {
+      throw new IllegalArgumentException("Rack names cannot be null");
+    }
 
     try (LoggingScope ls = loggingScopeFactory.evaluatorSubmit(req.getNumber())) {
-      final ResourceRequestEventImpl.Builder request = ResourceRequestEventImpl
+      final ResourceRequestEventImpl.Builder requestBuilder = ResourceRequestEventImpl
           .newBuilder()
           .setResourceCount(req.getNumber())
           .setVirtualCores(req.getNumberOfCores())
           .setMemorySize(req.getMegaBytes());
 
-      final ResourceCatalog.Descriptor descriptor = req.getDescriptor();
-      if (descriptor != null) {
-        if (descriptor instanceof RackDescriptor) {
-          request.addRackName(descriptor.getName());
-        } else if (descriptor instanceof NodeDescriptor) {
-          request.addNodeName(descriptor.getName());
-        } else {
-          throw new IllegalArgumentException("Unable to operate on descriptors of type " + descriptor.getClass().getName());
-        }
+      for (final String nodeName : req.getNodeNames()) {
+        requestBuilder.addNodeName(nodeName);
```

Comment:
can't the builder have an `addAll` method? that way the whole thing here could be a single statement.